### ISSUE TITLE
Cannot find High Charts Module

### ISSUE THREAD

**Issue by Maaster** [LeaPhant/flowabot#16]
Hey there, followed the installation guide and installed everything beforehand.
On doing 
>npm i 

it throws an error stating that the High Charts Module cannot be found.

Log of it: https://pastebin.com/MStvGdgM

I probably missed a really obvious thing and will feel stupid once its fixed, but I guess Id ask because I really wanna use this awesome bot.

For reference, running this in a completely new Ubuntu 18.04.2 VM.

Thanks :D

**Comment by LeaPhant** [comment]
Hi! There might be a problem with the interaction required during the highcharts installation. Try running `ACCEPT_HIGHCHARTS_LICENSE=1 npm i` instead.

**Comment by Maaster** [comment]
Fails with the same error sadly.

**Comment by LeaPhant** [comment]
That's really weird, does just installing highcharts on its own work? `npm i highcharts-export-server`

If it doesn't it's probably an issue with highcharts.

**Comment by Maaster** [comment]
This throws some more warnings beforehand, but results in the same error in the end.

https://puu.sh/DSo21/ac943b2403.png

**Comment by Maaster** [comment]
I installed nodejs via
>curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -sudo apt-get install -y nodejs

As per https://github.com/nodesource/distributions/blob/master/README.md

Maybe I was wrong there?

**Comment by LeaPhant** [comment]
No, that's the correct one. Can you please try again after removing `node_modules` by running `rm -rf node_modules`?

**Comment by Maaster** [comment]
Same error.

https://puu.sh/DSobj/ab72f114e9.png

**Comment by LeaPhant** [comment]
I've really run out of ideas then. :(

Maybe ask at Highcharts' issues tracker: https://github.com/highcharts/node-export-server/issues

**Comment by Maaster** [comment]
Alright, I nuked my VM and reinstalled everything.
By googling a bit it seems to be a problem of npm with phantomJS (see https://github.com/Alex-Rose/fb-messenger-cli/issues/109), as that module threw an error when trying to manually install highcharts-export-server.
With --ignore-scripts npm i runs through and I could configure everything.
However on starting, the canvas module is not found:
https://puu.sh/DSpZg/83c3aed9aa.png

I did
>apt-get install build-essential libcairo2-dev libpango1.0-dev libjpeg-dev libgif-dev librsvg2-dev

before and it went through with no errors.
This doesnt seem related to highcharts so Im guessing (or rather hoping) the --ignore-scripts parameter didnt break anything?

**Comment by LeaPhant** [comment]
Running with --ignore-scripts will not run any build scripts, so the native part of node-canvas is never built. It might also break other modules without notice so it really shouldn't be used. If you are on a Windows 10 host you could try using WSL by following the guide in #9 instead of a VM.

**Comment by Maaster** [comment]
Na das war ja ne schwere Geburt...

Anyway, it works now.
I can't fully put my finger on what it was now, but heres what I did:

>npm install -g node-pre-gyp

>npm install canvas

Then it threw an error regarding "primordials not defined", which google said was an node error in Version 12.
Which was...weird, as I installed 10 (see answer above).
Curiously node -v returned 12.
So I installed nvm, switch it to 10.16.0 and voila, npm i went through and so did npm start.

Bot is running on discord server now, thanks :)

Onto getting the pp command running, but thats for another issue.
